CCDB/include/CCDB/BasicCCDBManager.h

```diff
@@ -137,12 +147,17 @@ T* BasicCCDBManager::getForTimeStamp(std::string const& path, long timestamp)
                                                  mCreatedNotBefore ? std::to_string(mCreatedNotBefore) : "");
   }
   auto& cached = mCache[path];
+  if (mCheckObjValidityEnabled && cached.isValid(timestamp))
+    return reinterpret_cast<T*>(cached.objPtr.get());
+
   T* ptr = mCCDBAccessor.retrieveFromTFileAny<T>(path, mMetaData, timestamp, &mHeaders, cached.uuid,
                                                  mCreatedNotAfter ? std::to_string(mCreatedNotAfter) : "",
                                                  mCreatedNotBefore ? std::to_string(mCreatedNotBefore) : "");
   if (ptr) { // new object was shipped, old one (if any) is not valid anymore
     cached.objPtr.reset(ptr);
     cached.uuid = mHeaders["ETag"];
+    cached.startvalidity = std::stol(mMetaData["Valid-From"]);
```

Comment:
The CI tests are failing with ``std::invalid_argument: stol`` which means that ``mMetaData["Valid-From"]`` contains non-number. Could you debug the test or to run it with some logging to check what is the problem.

Reply:
Hi, sure thing, I'm on it

Reply:
It should be fine now, I used the Metadata instead of Headers, it is now fixed